pkg/controllers/disruption/consolidation_test.go

```diff
@@ -608,69 +623,112 @@ var _ = Describe("Consolidation", func() {
-			Expect(scheduling.NewNodeSelectorRequirements(nodeClaims[0].Spec.Requirements...).Has(v1.LabelInstanceTypeStable)).To(BeTrue())
-			Expect(scheduling.NewNodeSelectorRequirements(nodeClaims[0].Spec.Requirements...).Get(v1.LabelInstanceTypeStable).Has(mostExpensiveInstance.Name)).To(BeFalse())
+				// Expect that the new nodeclaim does not request the most expensive instance type
+				Expect(nodeClaims[0].Name).ToNot(Equal(nodeClaim.Name))
+				Expect(scheduling.NewNodeSelectorRequirements(nodeClaims[0].Spec.Requirements...).Has(v1.LabelInstanceTypeStable)).To(BeTrue())
+				Expect(scheduling.NewNodeSelectorRequirements(nodeClaims[0].Spec.Requirements...).Get(v1.LabelInstanceTypeStable).Has(mostExpensiveInstance.Name)).To(BeFalse())
 
-			// and delete the old one
-			ExpectNotFound(ctx, env.Client, nodeClaim, node)
-		})
-		It("can replace nodes if another nodePool returns no instance types", func() {
+				// and delete the old one
+				ExpectNotFound(ctx, env.Client, nodeClaim, node)
+			},
+			Entry("if the candidate is on-demand node", false),
+			Entry("if the candidate is spot node", true),
+		)
+		It("cannot replace spot node if less than minimum InstanceTypes flexibility", func() {
+			// Forcefully shrink the possible instanceTypes to be lower than 15 to replace a nodeclaim
+			cloudProvider.InstanceTypes = lo.Slice(fake.InstanceTypesAssorted(), 0, 5)
+			// Forcefully assign lowest possible instancePrice to make sure we have atleast one instance
+			// that is lower than the current node.
+			cloudProvider.InstanceTypes[0].Offerings[0].Price = 0.001
+			cloudProvider.InstanceTypes[0].Offerings[0].CapacityType = v1beta1.CapacityTypeSpot
+			spotInstances = lo.Filter(cloudProvider.InstanceTypes, func(i *cloudprovider.InstanceType, _ int) bool {
+				for _, o := range i.Offerings.Available() {
+					if o.CapacityType == v1beta1.CapacityTypeSpot {
+						return true
+					}
+				}
+				return false
+			})
+			// Sort the spot instances by pricing from low to high
+			sort.Slice(spotInstances, func(i, j int) bool {
+				return spotInstances[i].Offerings.Cheapest().Price < spotInstances[j].Offerings.Cheapest().Price
+			})
+			mostExpSpotInstance := spotInstances[len(spotInstances)-1]
+			mostExpSpotOffering := mostExpSpotInstance.Offerings[0]
 			labels := map[string]string{
 				"app": "test",
 			}
-			// create our RS so we can link a pod to it
+			spotNodeClaim, spotNode = test.NodeClaimAndNode(v1beta1.NodeClaim{
+				ObjectMeta: metav1.ObjectMeta{
+					Labels: map[string]string{
+						v1beta1.NodePoolLabelKey:     nodePool.Name,
+						v1.LabelInstanceTypeStable:   mostExpSpotInstance.Name,
+						v1beta1.CapacityTypeLabelKey: mostExpSpotOffering.CapacityType,
+						v1.LabelTopologyZone:         mostExpSpotOffering.Zone,
+					},
+				},
+				Status: v1beta1.NodeClaimStatus{
+					ProviderID:  test.RandomProviderID(),
+					Allocatable: map[v1.ResourceName]resource.Quantity{v1.ResourceCPU: resource.MustParse("10")},
+				},
+			})
```

Comment:
I believe you're redefining this because the instance types are different. You could do this with a 
`spotNodeClaim.Labels = lo.Assign(spotNodeClaim.Labels, map[string]string{ labels_you_want_to_change` and the same for the node, that way we are sure we're not overriding any other details of the nodeclaim or node specifically during this test.

Reply:
Addressed.